### PR TITLE
Contextual varargs parameters

### LINKED ISSUE #18090: Contextual vararg parameters
## Compiler version

3.3.2-RC1-bin-20230627-588a0b1-NIGHTLY and before

## Minimized code

```scala
def foo(using xs: Int*) = xs
val f = foo
```

## Output
```scala
[error] UsingVarargs.scala:2:12
[error] no implicit argument of type Int* was found for parameter xs of method foo
[error] val f = foo
[error]            ^
```

## Expectation

There seems to be no way to define an instance of a vararg type, e.g. `given Int* = 0` won't work.
Even if one defines a given instance of a corresponding non-vararg type or a type wrapped into a `Seq` (`given Int = 0` or `given Seq[Int] = Seq(0)`) it won't be picked by the implicits resolution. The only way to pass such parameter seems to be `foo(using 0)`.
Taking that into account it seems reasonable to raise a warning when a user defines a `using` parameter with a vararg type.

### PR DESCRIPTION
**Pull request by iusildra**
fixes #18090

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/reporting/messages.scala [hunk 1]
```diff
@@ -1319,6 +1319,14 @@ extends SyntaxMsg(VarArgsParamMustComeLastID) {
         |"""
 }
 
+class VarArgsParamCannotBeGiven(isGiven: Boolean)(using Context)
+extends SyntaxMsg(VarArgsParamCannotBeGivenID) {
+  def msg(using Context) = i"repeated parameters are not allowed in a ${if isGiven then "using" else "implicit"} clause"
+  def explain(using Context) =
+    "It is not possible to define a given with a repeated parameter type."
```
Comment (nicolasstucki) [inline compiler/src/dotty/tools/dotc/reporting/messages.scala:1326]:
```suggestion
    "It is not possible to define a given with a repeated parameter type. This hypothetical given parameter could always be satisfied by providing 0 arguments, which defeats the purpose of a given argument."
```

**Review by nicolasstucki — COMMENTED** [review]
Otherwise LGTM

**Review by nicolasstucki — APPROVED** [review]

**Comment by nicolasstucki** [comment]
@iusildra could you squash the commits?

File: tests/neg/i18090.check
Comment (nicolasstucki) [inline tests/neg/i18090.check:1]: This check file does not match anymore. Maybe a conflict with another PR. We should rebase and generate the file again.

Comment (nicolasstucki) [reply tests/neg/i18090.check:1]: Now it should be `[E188]`.

Comment (iusildra) [reply tests/neg/i18090.check:1]: ah, yes, I forgot to change the error code there too

**Review by nicolasstucki — APPROVED** [review]

**Pull request merged**